bin/run_demo.sh

```diff
@@ -1,26 +1,40 @@
 #!/bin/bash
 
+export DYLD_LIBRARY_PATH=${LSST_LIBRARY_PATH}
+
 # Exit on command failure
 set -e
 
 # Echo commands
 set -x
 
-# 
+#
 # Doing this as a shell script instead of scons to make it
 # easier to read.
 
 if [ ! -f DATA_REPO/butler.yaml ]; then
-    makeButlerRepo.py DATA_REPO
+    butler create DATA_REPO
+    butler register-instrument DATA_REPO lsst.obs.subaru.HyperSuprimeCam
 fi
 
-bin/ingestExternalData.py DATA_REPO $PWD/export_dir/export.yaml
+# Hack assuming posix datastore
+if [ ! -d DATA_REPO/HSC/calib ]; then
+    butler import DATA_REPO "${PWD}/export_dir" --export-file "${PWD}/export_dir/export.yaml" --output-run shared/ci_hsc --skip-dimensions instrument,physical_filter,detector
+fi
 
 # ingestRaws.py doesn't search recursively; over-specifying to work around that.
-bin/ingestRaws.py DATA_REPO export_dir/raw/hsc/raw/r/HSC-R/903342/
+if [ -z "$(find -L DATA_REPO/HSC/raw -type f)" ]; then
+    butler ingest-raws DATA_REPO --dir export_dir/HSC/raw/all/raw/r/HSC-R/
+    butler define-visits DATA_REPO -i HSC --collections HSC/raw/all
+fi
+
+export OMP_NUM_THREADS=1  # Disable in process threading
 
-# originally: -i calib/hsc,raw/hsc,masks/hsc,ref_cats,skymaps,shared/ci_hsc 
-pipetask run -d "visit=903342 AND detector=10" -j 1 -b DATA_REPO/butler.yaml \
-    -i calib/hsc,raw/hsc,ref_cats,shared/ci_hsc \
-    --register-dataset-types -p $PIPE_TASKS_DIR/pipelines/ProcessCcd.yaml \
-    --instrument lsst.obs.subaru.HyperSuprimeCam -o demo_collection
+# I dropped the masks/hsc and skymaps collections; apparently they weren't
+# needed for this graph so they weren't exported?
+# Also this will fail upon re-running, because it wants `--extend-run` if and
+# only if the output run already exists.
+pipetask run -d "exposure=903342 AND detector=10" -j 1 -b DATA_REPO/butler.yaml \
+    -i HSC/calib,HSC/raw/all,ref_cats,shared/ci_hsc \
+    --register-dataset-types -p "${PIPE_TASKS_DIR}/pipelines/ProcessCcd.yaml" \
+    --instrument lsst.obs.subaru.HyperSuprimeCam --output-run demo_collection
```

Comment:
Per the ticket, this should “run a single image and confirm that the results are generally sensible”. I don't see the second part of that here.

I note that we discussed on the ticket whether checking for exactly numerical tolerances was actually required; I _still_ think it would be useful if possible, but I'm not going to push for it. However, I think it is necessary to do something to check that the above actually did something meaningful, and to print a nice message for the user telling them if everything is ok.

Reply:
As I said in a comment on the ticket, I would dearly love to punt the comparison code to someone else. Merging this branch at least gives us part one. I have many other tickets I'm meant to be working on and this one has already taken far too long for me. Is there code lying around that can be inserted here with minimal effort? I don't want to be rewriting the compare.py script from lsst_dm_stack_demo and working out how to dump the source catalog to ascii.

Reply:
I think it'd be ok to merge the current functionality, file a new ticket to request some sort of checking of the outputs, and claim DM-22577 as done.